torch/distributed/_composable/fsdp/_fsdp_param.py

```diff
@@ -199,7 +199,11 @@ def _init_sharded_param(self, param: nn.Parameter, device: torch.device):
                     "FSDP requires the DP and TP mesh to have the same parent mesh but got: \n"
                     f"DP's global mesh: {dp_global_mesh}\nTP's global mesh: {tp_global_mesh}"
                 )
-            self._global_mesh = dp_global_mesh
+            assert dp_mesh.mesh_dim_names is not None and tp_mesh.mesh_dim_names is not None, (
```

Comment:
This assert statement can be split into two for easy debugging if the invariant is invalid later.